### ISSUE TITLE
Load sample data - Login Failed Error

### ISSUE THREAD

**Issue by mwdarab** [MicrosoftDocs/sql-docs#3765]
For #4 under Linux, it fails with "Login failed for user 'sa'. Reason: The account is disabled.." When the BDC is created, the 'sa' account is disabled by default. I had to go enable it, then run #4 again successfully.

Can there be a note that warns the reader that they will have to enable the 'sa' account? Also, after the data is loaded, is it safe to disable the sa account again? It would be nice if all that information is available for the reader who is trying to implement this (most likely for the first time).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 16adfeea-f75b-bd99-b9ae-44fe77e9c8d1
* Version Independent ID: 2c6e8c0d-4764-e6a1-e20c-8014a06264dd
* Content: [Load sample data - SQL Server big data clusters](https://docs.microsoft.com/en-us/sql/big-data-cluster/tutorial-load-sample-data?view=sql-server-ver15#feedback)
* Content Source: [docs/big-data-cluster/tutorial-load-sample-data.md](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/big-data-cluster/tutorial-load-sample-data.md)
* Product: **sql**
* Technology: **big-data-cluster**
* GitHub Login: @MikeRayMSFT
* Microsoft Alias: **mikeray**

**Comment by WilliamAntonRohm** [comment]
@mwdarab -- Mohammad, thank you for your feedback.

@MikeRayMSFT -- Mike, please look into this issue 3765.

**Comment by mwdarab** [comment]
This is still an issue.

**Comment by MikeRayMSFT** [comment]
Checking with engineering partners. Thank you for following up.

**Comment by axo959** [comment]
I followed the 'single node kubeadm script' I get an error the sa login is disabled during step #4 as well.  I also tried to login in remotely (but to port 31433) and from an interactive bash shell in my master container (had to install sqlcmd).  How'd you connect to the master instance if the sa account is disabled?  (what account/password?)

**Comment by axo959** [comment]
At some point they create an 'admin' user for sql auth.  The 'admin' user will have the "sa" password you set when running setup-dbc.sh on Linux. (coming from the AZDATA_PASSWORD env variable)

Following the docs from the 'single node kubeadmin script' to 'Tutorial: Load sample data into a SQL Server big data cluster', you dont really catch the need to login with a username admin instead of sa to the master sql instance.  

Looks like the image used is:
mcr.microsoft.com/mssql/bdc/mssql-server:2019-GDR1-ubuntu-16.04

I dont think its the image, I'm guessing disabling sa is somewhere in the bdc configs.  Maybe documented somewhere, but not in the path I took.

Once you can get logged in, you can enable sa if you choose with:
ALTER LOGIN sa ENABLE;
GO

Hope that helps someone!

**Comment by MikeRayMSFT** [comment]
We do not recommend enabling the SA account. 

If you want to use basic username/password as authentication mechanism then set AZDATA_USERNAME and AZDATA_PASSWORD environment variables before executing the script. Otherwise the script will use integrated authentication to connect to SQL Server Master instance and Knox gateway. Also, DNS name should be specified for the endpoints in order to use integrated authentication.

**Comment by axo959** [comment]
Yeah, I wont be using kerberos at all.  This was just to spin up quickly and play with bid data cluster.  Ideally we like to use ServerSPN, but thats not supported by the ODBC driver and probably not ideal for containers anyways.  Looks like there are other requirements to use the integrated as well.  Thats just not a world I'm ever in, so first I've read on it.  Looking at integrated needing FQDN, I wasnt sure what it meant by svc?  svc/fqdn:port.  I'm guessing svc name is an attribute of SPN?

Was trying to see where I got the idea I needed to get sa to work for sql/basic auth.   I notice on this page in the description for the env variables, you guys mention sa is disabled by default as a security best practice. (which I admit I missed) https://docs.microsoft.com/en-us/sql/big-data-cluster/deployment-guidance?view=sql-server-ver15  

But further down the page in a purple-ish !NOTE box is a part that reads:
'For SQL Server master, username provisioned to be used with the above password is sa.'
Maybe thats where I got sa stuck in my head.

Basically just wanted to get this getting started lab up in a sandbox.  Not sure how many data type folks (dba, db dev, analysts, data scientists, etc) are going to have experience with kerberos auth.  So if sa is enabled on a sandbox vm that will run for 2-3 days with some ms test data in it, I wasnt really too worried.

Thanks for the reply.

**Comment by MikeRayMSFT** [comment]
@axo959 - thank you for the detailed feedback. I'm looping in the right colleagues.

I'm updating the note to remove the information about the `sa` account because it is disabled. The table under [Environment variables](https://docs.microsoft.com/sql/big-data-cluster/deployment-guidance?view=sql-server-ver15#env) explains this.

| Environment variable | Requirement |Description |
|---|---|---|
| `AZDATA_USERNAME` | Required |The username for SQL Server big data cluster administrator. A sysadmin login with the same name is created in SQL Server master instance. As a security best practice, `sa` account is disabled. |
| `AZDATA_PASSWORD` | Required |The password for the user accounts created above. Same password is used for the `root` user, used for securing Knox gateway and HDFS. |
